docs/content/guides/2.alokai-essentials/2.alokai-react-native/6.product-page.md

```diff
@@ -0,0 +1,311 @@
+
+We have successfully fetched the list of products from the SAP Commerce Cloud backend and displayed them in the UI. But, the UI is not looking good. Let's use pre-built UI component `ProductCard` from `components` folder to display the product details.
+
+### Using ProductCard Component
+
+We will use the `ProductCard` component to display the product details. Open the `components/ProductCard.tsx` file. As you can see, right now it doesn't accept any props. Let's add the props to the `ProductCard` component.
+
+Add the following code at the top of the `components/ProductCard.tsx` file:
+
+```tsx
+export default function ProductCard({
+  product,
+}: {
+  product: Product;
+}) {
+  return (
+    <Link href={{
+      pathname: "Product Details/[product_code]",
+      params: { product_code: product.code },
+    }}>
+      <View style={styles.container}>
+        <View style={styles.imageContainer}>
+          <Image source={{ uri: transformImageUrl(product.firstVariantImage!) }} style={styles.image} />
+        </View>
+        <View style={styles.detailsContainer}>
+          <Text style={styles.title}>{product.name}</Text>
+          <Text style={styles.price}>{product.price?.formattedValue}</Text>
+          <Pressable style={styles.addToCartButton}>
+            <FontAwesome name="cart-plus" size={24} color="#fff" />
+            <Text style={styles.addToCartButtonText}>Add to cart</Text>
+          </Pressable>
+        </View>
+      </View>
+    </Link>
+  );
+}
+```
+
+In the above code, we have added the `product` prop to the `ProductCard` component. And used various UI components like `View`, `Image`, `Text`, `Pressable`, and `FontAwesome` to display the product details.
+
+Now, let's use the `ProductCard` component to display the product details in the `ProductListingPage`. Replace the `Text` component with the `ProductCard` component in the `FlatList` component in the `app/(tabs)/index.tsx` file:
+
+```diff
+- renderItem={({ item }) => (
+-   <Text>{item.name}</Text>
+- )}
++ renderItem={({ item }) => (
++   <ProductCard product={item} />
++ )}
+```
+
+Now, save the changes and run the application. Now, when you visit the product listing page, you will see the list of products displayed using the `ProductCard` component.
+
+<img src="./images/plp-2.webp" alt="Product Listing Page" style="max-width: 350px;" />
+
+It's up to you to style the `ProductCard` component as you like, as well as to add more details to it and style the `ProductListingPage` as you like.
+
+Next, let's create a Product Details Page to display the product details.
+
+## Creating a Product Details Page
+
+To create a Product Details Page, we need to create a new dynamic route for the product details page. Create a new directory `app/Product Details/[product_code]` in the `app` directory. Inside the `app/Product Details/[product_code]` directory, create a new file `index.tsx` and add the following code:
```

Comment:
Can you explain the `Product Details` thing a little bit more? It looks weird, but I know this is what caused you pain so maybe you can share that knowledge here.